File: crates/filter-list-manager/src/manager/models/filter_list_metadata_with_body.rs
//! Metadata for a remote filter list.
use crate::FilterListMetadata;

/// Metadata (with body) for a remote filter list.
#[derive(Clone, Debug)]
pub struct FilterListMetadataWithBody {
    /// Metadata for a remote filter list.
    pub metadata: FilterListMetadata,
    /// Filter body.
    pub filter_body: String,
}

File: platform/windows/uniffi/AdGuard.FilterListManager/MarshalLogic/FilterListMetadataWithBody.cs
namespace AdGuard.FilterListManager.MarshalLogic
{
    /// <summary>
    /// Filter metadata holder
    /// </summary>
    public class FilterListMetadataWithBody
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="FilterListMetadataWithBody"/> class.
        /// </summary>
        /// <param name="metadata">The filter metadata.</param>
        /// <param name="filterBody">The filter body.</param>
        public FilterListMetadataWithBody(FilterListMetadata metadata, string filterBody)
        {
            this.Metadata = metadata;
            this.FilterBody = filterBody;
        }
        /// <summary>
        /// Gets or sets the filter metadata.
        /// </summary>
        public FilterListMetadata Metadata { get; set; }

        /// <summary>
        /// Gets or sets the filter body.
        /// </summary>
        public string FilterBody { get; set; }
    }
}